compiler/src/dmd/func.d

```diff
@@ -2033,6 +2034,7 @@ extern (C++) class FuncDeclaration : Declaration
                 return 0;
             }
         });
+        catch (Throwable) assert(0);
```

Comment:
it should only catch Exception

Reply:
Throwable influences the `nothrow` attribute. Besides, this is just temporary scaffolding.

Also, assert's can throw an Exception, and wouldn't want to catch them here.

Reply:
catching Exception satisfies nothrow. And asserts do not throw Exceptions and even fi they did, catch(Throwable) includes that anyway.